features/API/auth.feature

```diff
@@ -0,0 +1,48 @@
+@API @Auth
+Feature: Test an ability to register and login
+
+  Scenario: User is attempting to register using valid data
+    Given an user is requesting "/api/register" using "POST"
+    And request body contains "email" equal "testuser@example.com"
+    And request body contains "password" equal "password"
+    And request body contains "password_confirmation" equal "password"
+    And request body contains "name" equal "username"
+    When a request is sent
+    Then a response status code should be "200"
+
+  Scenario: User is attempting to register using existing email
+    Given an user is requesting "/api/register" using "POST"
+    And request body contains "email" equal "testuser@example.com"
+    And request body contains "password" equal "password"
+    And request body contains "password_confirmation" equal "password"
+    And request body contains "name" equal "username"
+    When a request is sent
+    Then a response status code should be "422"
```

Comment:
I don't get it. These two tests (:4 and :13) are almost identical and for some reason they're returning different things? Every test should be ran on tabula rasa; in other case, we won't be able to predict what happened between tests and what outcome would appear. In this case you should set up your environment:
```
    And there is an user created:
      | id                                   | email             |
      | 00000000-0000-0000-0000-000000000000 | testuser@example.com |
```

And of course refresh database (with `KrzysztofRewak\Larahat\Helpers\RefreshingDatabase` trait added to Context).

Reply:
(or just add UserContext to this suite)